### PR TITLE
Service

### PR DESCRIPTION
**Pull request by okellogabrielinnocent**
Add service controller
Create request when service is made to check if the user is authenticated and desired fields are made

### CODE REVIEW

File: app/Http/Controllers/Item/ItemController.php [hunk 1]
```diff
@@ -26,8 +26,8 @@ public function index()
         // TO-DO
         // Ask fro help from Jeseph of how to use the config values on pagination
         // get the pagination number or a default
-        $items = $this->$itemRepository ?? 10;
-        return view('items' . config('settings.pagination.small') . '.index')->with(['$items' => $items]);
+        $items = $this->$itemRepository->paginate(config('settings.pagination.small')) ?? 10;
```
Comment (jmatembu) [inline app/Http/Controllers/Item/ItemController.php]:
```suggestion
        $items = $this->$itemRepository->paginate(config('settings.pagination.small'));
```

Previously you could end up with `$items = 10`; But questions is, where are you getting the number 10 from? By having the value coming from the `settings.pagination.small` means you are already setting a default value.

File: app/Http/Controllers/Serivice/ServiceController.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        // TO-DO
+        // get the pagination number or a default to 10
+        $services = $this->$serviceRepository->paginate(config('settings.pagination.small')) ?? 10;
```
Comment (jmatembu) [inline app/Http/Controllers/Serivice/ServiceController.php]: See my previous comment on removing the value `10`.

File: app/Http/Controllers/Serivice/ServiceController.php [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+     */
+    public function destroy(Service $service)
+    {
+        try {
+            $service->delete();
+            return redirect()->to('services')->withSuccess('Service Deleted Succesfuly');
+        } catch (\Exception $e) {
+            // log errors the error in the system
+            \Log::debug($e->getMessage());
+            return "No Service to delete!";
```
Comment (jmatembu) [inline app/Http/Controllers/Serivice/ServiceController.php]: You need to redirect the user not just printing the message on the screen. That message should instead be shown on the screen (may be `services.index`) which you will redirect the user back to.

File: app/Http/Controllers/Serivice/ServiceRepository.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+
+namespace App\Http\Controllers\Service;
+
+use App\Service;
+use Illuminate\Database\Eloquent\Model;
+
+class ServiceRepository
```
Comment (jmatembu) [inline app/Http/Controllers/Serivice/ServiceRepository.php:9]: Add a method to delete a model. At the moment, in the controller you are not using your repository class to delete a model. You may need to make a similar change to other repositories as well.

**Review by jmatembu — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Serivice/ServiceController.php (hunk 3 above)
Comment (okellogabrielinnocent) [reply app/Http/Controllers/Serivice/ServiceController.php]: Thanks 👌

File: app/Http/Controllers/Serivice/ServiceController.php (hunk 2 above)
Comment (okellogabrielinnocent) [reply app/Http/Controllers/Serivice/ServiceController.php]: Have done it

**Comment by okellogabrielinnocent** [comment]
[Service](https://trello.com/c/QJPbLsPn/6-service)

**Pull request merged**